src/sidebar/components/ShareLinks.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  LinkBase,
+  EmailIcon,
+  SocialFacebookIcon,
+  SocialTwitterIcon,
+} from '@hypothesis/frontend-shared/lib/next';
+import type { IconComponent } from '@hypothesis/frontend-shared/lib/types';
+
+type ShareLinkProps = {
+  /** The SVG icon component to use for this link */
+  icon: IconComponent;
+
+  /** Accessible label/tooltip for link */
+  label: string;
+
+  /** URI for sharing this annotation */
+  uri: string;
+};
+
+/**
+ * A single sharing link as a list item
+ */
+function ShareLink({ label, icon: Icon, uri }: ShareLinkProps) {
+  return (
+    <li>
+      <LinkBase
+        aria-label={label}
+        classes="text-grey-6 hover:text-color-text block"
+        href={uri}
+        title={label}
+        target="_blank"
+      >
+        <Icon
+          // Make the icons sized to the current text size to allow for
+          // differently-sized sharing icon links
+          className="w-em h-em"
+        />
+      </LinkBase>
+    </li>
+  );
+}
+
+export type ShareLinksProps = {
+  shareURI: string;
+};
+
+/**
+ * A list of share links to social-media platforms.
+ */
+export default function ShareLinks({ shareURI }: ShareLinksProps) {
+  // This is the double-encoded format needed for other services (the entire
+  // URI needs to be encoded because it's used as the value of querystring params)
```

Comment:
I suggest we remove this comment. I think the reason why the URI needs to be encoded is obvious from the nearby context, and it isn't clear what "double-encoded" means.

What would be safer in general is to handle encoding as part of URL generation, eg. using a templated string like we do in the LMS with the `urlPath` helper. That's outside the scope of this PR though.